### PR TITLE
[BUGFIX] Use correct paths for CJS and ESM compatibility

### LINKED ISSUE #12: Package.json incorrectly formatted
I'm getting a failure on a Travis build from this update. Looks like you just added a dual-module package. With `type: module` in the package.json, I _think_ `main` should point to the ESM build or needs the `.cjs` extension.

Build is here: https://travis-ci.org/github/mAAdhaTTah/brookjs/jobs/679426871

I haven't written a dual module package, so I could be wrong on the fix, but the current implementation is def broken, unfortunately.

### PR DESCRIPTION
**Pull request by RyanZim**
Fixes #12

Introduced in feb90a40501c8ef69b0c65bdf1eb703182214407

Tested `npm pack` and `npm install` from tarball, to verify changes work with both ESM and CJS usage in a variety of Node versions.

### CODE REVIEW

**Comment by RyanZim** [comment]
CI is failing due to the CI running on Node 6, but the code using `async`/`await`. This was broken before this PR: https://travis-ci.org/github/then/is-promise/jobs/679428226

**Comment by RyanZim** [comment]
Added commit to downgrade mocha to the version that was used previously, to fix Node 6 tests.

**Comment by eykrehbein** [comment]
I think it'd be good to update the version in `package.json` and `component.json` as well. `2.2.1` would be the right one I guess

**Comment by RyanZim** [comment]
@eykrehbein that's the job of whoever is releasing the patch usually.

**Comment by RyanZim** [comment]
Attn: @ForbesLindesay @edef1c @mxstbr 

For clarity, v2.2.0 is currently broken on all modern Node versions that support ESM; regardless of whether the consumer is using the CJS or ESM version.

**Comment by szmarczak** [comment]
I'd adding Node.js 12, 13 and 14 to tests.

**Comment by szmarczak** [comment]
Node.js versions < 10 have been deprecated, I see no point for testing Node.js 6 and 8.

**Comment by RyanZim** [comment]
@szmarczak Should be done, but I'd prefer to keep this bugfix focused on fixing the actual bugs, without making other changes. Dropping support for Node 6 & 8 would be a breaking, semver-major change; we need a patch fix for the currently broken v2.x series first.

**Comment by dncrews** [comment]
Just in time for my production release to the company in 4 minutes lol

**Comment by kamescg** [comment]
For those using Yarn you can also add a resolution to force install a working version.

```
"resolutions": {
  "is-promise": "2.1.0"
},
```

**Comment by szmarczak** [comment]
> I'd prefer to keep this bugfix focused on fixing the actual bugs

Then you want to add just Node.js 10, 12, 13 and 14. The fix needs to be tested somehow.

File: package.json [hunk 1]
```diff
@@ -22,6 +22,6 @@
   "license": "MIT",
   "devDependencies": {
     "better-assert": "^1.0.2",
-    "mocha": "^7.1.1"
+    "mocha": "~1.7.4"
```
Comment (ljharb) [inline package.json:25]: not sure about this one

**Comment by sabinlungudotcpp** [comment]
> Attn: @ForbesLindesay @edef1c @mxstbr
> 
> For clarity, v2.2.0 is currently broken on all modern Node versions that support ESM; regardless of whether the consumer is using the CJS or ESM version.

So this error is normal then? I'm trying to run it with docker.

> frontend@0.1.0 start /app
> react-scripts start

internal/modules/cjs/loader.js:584
            if (e.code !== 'ERR_PACKAGE_PATH_NOT_EXPORTED') throw e;
                                                            ^

Error [ERR_INVALID_PACKAGE_TARGET]: Invalid "exports" main target "index.js" defined in the package config /app/node_modules/is-promise/package.json
    at resolveExportsTarget (internal/modules/cjs/loader.js:542:13)
    at resolveExportsTarget (internal/modules/cjs/loader.js:581:20)
    at applyExports (internal/modules/cjs/loader.js:455:14)
    at resolveExports (internal/modules/cjs/loader.js:508:23)
    at Function.Module._findPath (internal/modules/cjs/loader.js:632:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1001:27)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/app/node_modules/run-async/index.js:3:17) {
  code: 'ERR_INVALID_PACKAGE_TARGET'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! frontend@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the frontend@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-25T17_30_34_910Z-debug.log
sabin@Sabins-MacBook-Pro frontend %

**Comment by sabinlungudotcpp** [comment]
> frontend@0.1.0 start /app
> react-scripts start

Is this supposed to happen? I have downgraded node to version 12.12.0 and this still happens when I try to run it with docker.

internal/modules/cjs/loader.js:584
            if (e.code !== 'ERR_PACKAGE_PATH_NOT_EXPORTED') throw e;
                                                            ^

Error [ERR_INVALID_PACKAGE_TARGET]: Invalid "exports" main target "index.js" defined in the package config /app/node_modules/is-promise/package.json
    at resolveExportsTarget (internal/modules/cjs/loader.js:542:13)
    at resolveExportsTarget (internal/modules/cjs/loader.js:581:20)
    at applyExports (internal/modules/cjs/loader.js:455:14)
    at resolveExports (internal/modules/cjs/loader.js:508:23)
    at Function.Module._findPath (internal/modules/cjs/loader.js:632:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1001:27)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/app/node_modules/run-async/index.js:3:17) {
  code: 'ERR_INVALID_PACKAGE_TARGET'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! frontend@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the frontend@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-25T17_30_34_910Z-debug.log
sabin@Sabins-MacBook-Pro frontend %

File: package.json (hunk 1 above)
Comment (RyanZim) [reply package.json:25]: Downgraded to fix existing broken CI; due to mocha v7 not working on Node 6; which is still supported by the module. https://github.com/then/is-promise/pull/15#issuecomment-619403970

**Comment by ljharb** [comment]
@sabinlungudotcpp yes, v2.2.0 is broken. Pin to v2.1 for now.

File: package.json (hunk 1 above)
Comment (ljharb) [reply package.json:25]: ah - mocha 3 should still work, it's mocha 4 where it stopped being useful for things that like back compat :-)

**Comment by sabinlungudotcpp** [comment]
> @sabinlungudotcpp yes, v2.2.0 is broken. Pin to v2.1 for now.

2.2.0 or do you mean 12.2.0?

**Comment by RyanZim** [comment]
> > @sabinlungudotcpp yes, v2.2.0 is broken. Pin to v2.1 for now.
> 
> 2.2.0 or do you mean 12.12.0?

2.2.0 is broken, pin to 2.1.0

**Comment by sabinlungudotcpp** [comment]
> > > @sabinlungudotcpp yes, v2.2.0 is broken. Pin to v2.1 for now.
> > 
> > 
> > 2.2.0 or do you mean 12.12.0?
> 
> 2.2.0 is broken, pin to 2.1.0

What's the command to downgrade to 2.1.0?

**Comment by byaussy** [comment]
> > > > @sabinlungudotcpp yes, v2.2.0 is broken. Pin to v2.1 for now.
> > > 
> > > 
> > > 2.2.0 or do you mean 12.12.0?
> > 
> > 
> > 2.2.0 is broken, pin to 2.1.0
> 
> What's the command to downgrade to 2.1.0?

npm install is-promise@2.1.0

**Comment by sabinlungudotcpp** [comment]
> npm install is-promise@2.1.0

Thanks, and what is the command to verify the version of it just to be sure it's installed and I'm using it?

**Comment by byaussy** [comment]
> > npm install is-promise@2.1.0
> 
> Thanks, and what is the command to verify the version of it just to be sure it's installed and I'm using it?

npm list is-promise

**Comment by sabinlungudotcpp** [comment]
> > > npm install is-promise@2.1.0
> > 
> > 
> > Thanks, and what is the command to verify the version of it just to be sure it's installed and I'm using it?
> 
> npm list is-promise

Thank you so much, it's working now!

**Comment by fisker** [comment]
Why can't we just unpublish `is-promise@2.2.0` first? It's not over 24h.

**Comment by RyanZim** [comment]
@fisker We are all waiting for someone that has publish access to the package on npm to notice this.

**Comment by kolaval** [comment]
Clean install of node v12.16.2, npm 6.14.4
Cannot install vue:  `npm install -g @vue/cli`

**Comment by ForbesLindesay** [comment]
Check it is using v2.2.2 of is-promise as v2.2.1 still had some issues

**Pull request merged**